File: aforclinic-web-main/aforclinic-web-main/shared/types/Popup.ts
export interface Popup {
  /**
   * 팝업 식별자
   */
  id: string;
  /**
   * 팝업 타입
   */
  type: 'CAROUSEL' | 'MODAL';
  /**
   * 팝업 제목
   */
  title: string;
  /**
   * 이미지 식별자
   */
  imageId: string;
  /**
   * 링크
   */
  linkUri?: string;
  /**
   * TOP
   */
  positionTop?: number;
  /**
   * LEFT
   */
  positionLeft?: number;
}

File: aforclinic-api-main/aforclinic-api-main/src/main/java/com/xenialsoft/api/core/popup/model/Popup.java
package com.xenialsoft.api.core.popup.model;

import java.time.LocalDate;

import com.xenialsoft.api.common.Auditable;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

/**
 * 팝업 정보를 저장하는 엔티티 클래스.
 *
 * @author xenialsoft
 * @since 
 */
@Getter
@Setter
@NoArgsConstructor
public class Popup extends Auditable {
    
    /**
     * 팝업 고유 식별자
     */
    private String id;
    
    /**
     * 팝업 유형
     */
    private String type;
    
    /**
     * 화면 상단으로부터의 거리(px)
     */
    private int positionTop;
    
    /**
     * 화면 좌측으로부터의 거리(px)
     */
    private int positionLeft;
    
    /**
     * 제목
     */
    private String title;
    
    /**
     * 파일 고유 식별자
     */
    private String imageId;
    
    /**
     * 링크
     */
    private String linkUri;
    
    /**
     * 팝업 시작일
     */
    private LocalDate startDate;
    
    /**
     * 팝업 종료일
     */
    private LocalDate endDate;
    
    /**
     * 화면 표시 순서
     */
    private int displayOrder;
    
    /**
     * 상태
     */
    private PopupStatus status;
    
}
